openssl-sys/src/hpke.rs

```diff
@@ -0,0 +1,58 @@
+#[cfg(ossl320)]
+use crate::OSSL_HPKE_SUITE;
+use libc::c_int;
+
+#[cfg(ossl320)]
+pub const OSSL_HPKE_MODE_BASE: c_int = 0x00;
+#[cfg(ossl320)]
+pub const OSSL_HPKE_MODE_PSK: c_int = 0x01;
+#[cfg(ossl320)]
+pub const OSSL_HPKE_MODE_AUTH: c_int = 0x02;
+#[cfg(ossl320)]
+pub const OSSL_HPKE_MODE_PSKAUTH: c_int = 0x03;
+
+#[cfg(ossl320)]
+pub const OSSL_HPKE_ROLE_SENDER: c_int = 0x00;
+#[cfg(ossl320)]
+pub const OSSL_HPKE_ROLE_RECEIVER: c_int = 0x01;
+
+#[cfg(ossl320)]
+pub const OSSL_HPKE_KEM_ID_P256: u16 = 0x10;
+#[cfg(ossl320)]
+pub const OSSL_HPKE_KEM_ID_P384: u16 = 0x11;
+#[cfg(ossl320)]
+pub const OSSL_HPKE_KEM_ID_P521: u16 = 0x12;
+#[cfg(ossl320)]
+pub const OSSL_HPKE_KEM_ID_X25519: u16 = 0x20;
+#[cfg(ossl320)]
+pub const OSSL_HPKE_KEM_ID_X448: u16 = 0x21;
+
+#[cfg(ossl320)]
+pub const OSSL_HPKE_KDF_ID_HKDF_SHA256: u16 = 0x01;
+#[cfg(ossl320)]
+pub const OSSL_HPKE_KDF_ID_HKDF_SHA384: u16 = 0x02;
+#[cfg(ossl320)]
+pub const OSSL_HPKE_KDF_ID_HKDF_SHA512: u16 = 0x03;
+
+#[cfg(ossl320)]
+pub const OSSL_HPKE_AEAD_ID_AES_GCM_128: u16 = 0x01;
+#[cfg(ossl320)]
+pub const OSSL_HPKE_AEAD_ID_AES_GCM_256: u16 = 0x02;
+#[cfg(ossl320)]
+pub const OSSL_HPKE_AEAD_ID_CHACHA_POLY1305: u16 = 0x03;
+#[cfg(ossl320)]
+pub const OSSL_HPKE_AEAD_ID_EXPORTONLY: u16 = 0xFFFF;
+
+#[cfg(all(ossl320, not(osslconf = "OPENSSL_NO_ECX")))]
+pub const OSSL_HPKE_SUITE_DEFAULT: OSSL_HPKE_SUITE = OSSL_HPKE_SUITE {
+    kem_id: OSSL_HPKE_KEM_ID_X25519,
+    kdf_id: OSSL_HPKE_KDF_ID_HKDF_SHA256,
+    aead_id: OSSL_HPKE_AEAD_ID_AES_GCM_128,
+};
+
+#[cfg(all(ossl320, osslconf = "OPENSSL_NO_ECX"))]
```

Comment:
```suggestion
#[cfg(all(ossl320, osslconf = "OPENSSL_NO_EC"))]
```

Reply:
I'm just echoing the hpke header by using OPENSSL_NO_ECX here: https://github.com/openssl/openssl/blob/8ad98cce41aa8a6278f7ade6ad2f70b80b194b72/include/openssl/hpke.h#L90

Reply:
ECX is for Curve25519, but you're guarding P-256 here, which [isn't available][1] when NO_EC is set. The header makes no sense as it is.

https://github.com/openssl/openssl/blob/8ad98cce41aa8a6278f7ade6ad2f70b80b194b72/crypto/hpke/hpke_util.c#L64